finality-aleph/src/aggregation/aggregator.rs

```diff
@@ -75,6 +82,32 @@ impl<H: Copy + Hash, PMS> BlockSignatureAggregator<H, PMS> {
             None => Err(AggregatorError::NoHashFound),
         }
     }
+
+    pub fn status_report(&self) {
+        let mut status = String::from("Block Signature Aggregator status report: ");
+
+        status.push_str(&format!(
+            "started hashes - {:?}; ",
+            self.started_hashes.len()
+        ));
+
+        status.push_str(&format!(
+            "collected sginatures - {:?}; ",
+            self.signatures.len()
+        ));
+
+        status.push_str(&format!("hashes in queue - {:?}; ", self.hash_queue.len()));
+
+        if let Some(hash) = self.hash_queue.front() {
+            status.push_str(&format!(
+                "front of hash queue - {} for - {:?}; ",
+                hash,
+                Instant::now() - self.front_instant
```

Comment:
This code might panic -- avoid unchecked arithmetic on instants, because monotonicity might fail. Use `saturating_duration_since().as_f64`.  Please, print it with precision of `2` decimal places.